src/coreComponents/common/logger/ErrorHandling.cpp

```diff
@@ -0,0 +1,252 @@
+
+namespace geos
+{
+static constexpr std::string_view g_level1Start = "  - ";
+static constexpr std::string_view g_level1Next =  "    ";
+static constexpr std::string_view g_level2Start = "    - ";
+static constexpr std::string_view g_level2Next =  "      ";
+static constexpr std::string_view g_level3Start = "      - ";
+static constexpr std::string_view g_level3Next =  "        ";
+static constexpr std::string_view g_callStackMessage =
+  "Callstack could not be retrieved. The format does not match the expected one.";
+
+ErrorLogger g_errorLogger{};
+
+void ErrorLogger::createFile()
+{
+  if( stringutilities::endsWith( m_filename, ".yaml" ) )
+  {
+    std::ofstream yamlFile( std::string( m_filename ), std::ios::out );
+    if( yamlFile.is_open() )
+    {
+      yamlFile << "errors: \n\n";
+      yamlFile.close();
+    }
+    else
+    {
+      GEOS_LOG_RANK( GEOS_FMT( "Unable to open error file for writing: {}", m_filename ) );
+    }
+  }
+  else
+  {
+    enableFileOutput( false );
+    GEOS_LOG_RANK( GEOS_FMT( "{} is a bad file name argument. The file must be in yaml format.", m_filename ) );
+  }
+}
+
+std::string ErrorLogger::ErrorContext::attributeToString( ErrorLogger::ErrorContext::Attribute attribute )
+{
+  switch( attribute )
+  {
+    case ErrorLogger::ErrorContext::Attribute::InputFile: return "inputFile";
+    case ErrorLogger::ErrorContext::Attribute::InputLine: return "inputLine";
+    case ErrorLogger::ErrorContext::Attribute::DataPath: return "dataPath";
+    default: return "Unknown";
+  }
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::addToMsg( std::exception const & e, bool toEnd )
+{
+  if( toEnd )
+  {
+    m_msg = m_msg + e.what();
+  }
+  else
+  {
+    m_msg = e.what() + m_msg;
+  }
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::addToMsg( std::string_view errorMsg, bool toEnd )
+{
+  if( toEnd )
+  {
+    m_msg = m_msg + std::string( errorMsg );
+  }
+  else
+  {
+    m_msg = std::string( errorMsg ) + m_msg;
+  }
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::setCodeLocation( std::string_view msgFile, integer msgLine )
+{
+  m_file = msgFile;
+  m_line = msgLine;
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::setType( ErrorLogger::MsgType msgType )
+{
+  m_type = msgType;
+  return *this;
+}
+
+void ErrorLogger::ErrorMsg::addContextInfoImpl( ErrorLogger::ErrorContext && ctxInfo )
+{
+  m_contextsInfo.emplace_back( std::move( ctxInfo ) );
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::setRank( int rank )
+{
+  m_ranksInfo.push_back( rank );
+  return *this;
+}
+
+ErrorLogger::ErrorMsg & ErrorLogger::ErrorMsg::addCallStackInfo( std::string_view ossStackTrace )
+{
+  std::string str = std::string( ossStackTrace );
+  std::istringstream iss( str );
+  std::string stackLine;
+  std::size_t index;
+
+  std::regex pattern( R"(Frame \d+: \S+)" );
+
+  while( std::getline( iss, stackLine ) )
+  {
+    if( std::regex_search( stackLine, pattern ))
```

Comment:
If the stack trace doesn't match the expected format, I think we shouldn't just throw it away. We should just show the raw text.